src/pages/pass-finalisation/cat-d/_tests_/pass-finalisation.cat-d.page.spec.ts

```diff
@@ -0,0 +1,346 @@
+      automaticBanner: true,
+      manualBanner: false,
+      desc: 'Automatic banner shown when automatic transmission and code78 present',
+    },
+    {
+      category: TestCategory.DE,
+      code78: false,
+      transmission: TransmissionType.Automatic,
+      automaticBanner: false,
+      manualBanner: true,
+      desc: 'Manual banner shown when automatic transmission and no code78 present',
+    },
+    {
+      category: TestCategory.DE,
+      code78: false,
+      transmission: TransmissionType.Manual,
+      automaticBanner: false,
+      manualBanner: true,
+      desc: 'Manual banner shown when manual transmission and no code78 present',
+    },
+    {
+      category: TestCategory.DE,
+      code78: true,
+      transmission: TransmissionType.Manual,
+      automaticBanner: false,
+      manualBanner: true,
+      desc: 'Manual banner shown when manual transmission and code78 present',
+    },
+    {
+      category: TestCategory.D1,
+      code78: true,
+      transmission: TransmissionType.Automatic,
+      automaticBanner: false,
+      manualBanner: false,
+      desc: 'No banner shown when automatic transmission and code78 present',
+    },
+    {
+      category: TestCategory.D1,
+      code78: false,
+      transmission: TransmissionType.Automatic,
+      automaticBanner: false,
+      manualBanner: false,
+      desc: 'No banner shown when automatic transmission and no code78 present',
+    },
+    {
+      category: TestCategory.D1,
+      code78: false,
+      transmission: TransmissionType.Manual,
+      automaticBanner: false,
+      manualBanner: false,
+      desc: 'No banner shown when manual transmission and no code78 present',
+    },
+    {
+      category: TestCategory.D1,
+      code78: true,
+      transmission: TransmissionType.Manual,
+      automaticBanner: false,
+      manualBanner: false,
+      desc: 'No banner shown when manual transmission and code78 present',
+    },
+    {
+      category: TestCategory.D1E,
+      code78: true,
+      transmission: TransmissionType.Automatic,
+      automaticBanner: false,
+      manualBanner: false,
+      desc: 'No banner shown when automatic transmission and code78 present',
+    },
+    {
+      category: TestCategory.D1E,
+      code78: false,
+      transmission: TransmissionType.Automatic,
+      automaticBanner: false,
+      manualBanner: false,
+      desc: 'No banner shown when automatic transmission and no code78 present',
+    },
+    {
+      category: TestCategory.D1E,
+      code78: false,
+      transmission: TransmissionType.Manual,
+      automaticBanner: false,
+      manualBanner: false,
+      desc: 'No banner shown when manual transmission and no code78 present',
+    },
+    {
+      category: TestCategory.D1E,
+      code78: true,
+      transmission: TransmissionType.Manual,
+      automaticBanner: false,
+      manualBanner: false,
+      desc: 'No banner shown when manual transmission and code78 present',
+    },
+  ];
```

Comment:
I'm sure the data conditioning is the same across the other categories but I would consider moving this out to separate file - maybe in a mocks folder.